lib/CL/devices/ventus/pocl_ventus.cc

```diff
@@ -1165,45 +1165,45 @@ pocl_ventus_driver_copy (void *data, pocl_mem_identifier *dst_mem_id, cl_mem dst
     free(host_ptr);
 }
 
-void
-pocl_ventus_read_rect (void *data, void *__restrict__ const host_ptr,
-                       pocl_mem_identifier *src_mem_id, cl_mem src_buf,
-                       const size_t *__restrict__ const buffer_origin,
-                       const size_t *__restrict__ const host_origin,
-                       const size_t *__restrict__ const region,
-                       size_t const buffer_row_pitch,
-                       size_t const buffer_slice_pitch,
-                       size_t const host_row_pitch,
-                       size_t const host_slice_pitch)
+void 
+pocl_ventus_read_rect(void *data, void *__restrict__ const host_ptr,
+                           pocl_mem_identifier *src_mem_id, cl_mem src_buf,
+                           const size_t *__restrict__ const buffer_origin,
+                           const size_t *__restrict__ const host_origin,
+                           const size_t *__restrict__ const region,
+                           size_t const buffer_row_pitch,
+                           size_t const buffer_slice_pitch,
+                           size_t const host_row_pitch,
+                           size_t const host_slice_pitch)
 {
-    cl_mem_flags flags = src_buf->flags;
-    void *__restrict__ device_ptr = src_mem_id->mem_ptr;
+  vt_device_data_t *d = (vt_device_data_t *)data;
+  void *__restrict__ device_ptr = src_mem_id->mem_ptr;
+  uint64_t src_addr = (uint64_t)(*(uint64_t *)device_ptr);
 
-    if (flags & CL_MEM_USE_HOST_PTR) {
```

Comment:
Why do you want to remove this case?

Reply:
The ventus architecture does not support CL_MEM_USE_HOST_PTR yet, so use copy instead.